extensions/VisualEditor/wikia/modules/ve/ui/dialogs/ve.ui.WikiaSourceModeDialog.js

```diff
@@ -0,0 +1,216 @@
+		this.sourceModeTextarea.$input.css( {
+			'z-index': 9999999,
+			position: 'relative'
+		} );
+		mw.loader.using(
+			'ext.wikia.LinkSuggest',
+			function () {
+				this.sourceModeTextarea.$input.linksuggest();
+			}.bind( this )
+		);
+	}
+};
+
+/**
+ * @inheritdoc
+ */
+ve.ui.WikiaSourceModeDialog.prototype.getSetupProcess = function ( data ) {
+	return ve.ui.WikiaSourceModeDialog.super.prototype.getSetupProcess.call( this, data )
+		.first( function () {
+			this.surface = data.surface;
+			this.openCount++;
+			this.timings.serializeStart = ve.now();
+		}, this )
+		.next( function () {
+			this.$content.startThrobbing();
+			this.surface.getTarget().serialize(
+				ve.dm.converter.getDomFromModel( this.getFragment().getDocument(), false ),
+				this.onSerialize.bind( this )
+			);
+		}, this );
+};
+
+/**
+ * @method
+ * @param {string} wikitext Wikitext returned from Parsoid.
+ */
+ve.ui.WikiaSourceModeDialog.prototype.onSerialize = function ( wikitext ) {
+	this.sourceModeTextarea.setValue( wikitext );
+	this.sourceModeTextarea.$input.focus();
+	this.sourceModeTextarea.$input.scrollTop( 0 );
+	this.sourceModeTextarea.$input[0].setSelectionRange( 0, 0 );
+	this.$content.stopThrobbing();
+
+  // ve-upstream-sync - review - @author: Paul Oslund
+	// ve.track( 'wikia', {
+	// 	action: ve.track.actions.SUCCESS,
+	// 	label: 'dialog-source-serialize',
+	// 	value: ve.now() - this.timings.serializeStart
+	// } );
+};
+
+ve.ui.WikiaSourceModeDialog.prototype.getActionProcess = function ( action ) {
+	if ( action === 'apply' ) {
+		return new OO.ui.Process( function () {
+			this.timings.parseStart = ve.now();
+			this.$content.startThrobbing();
+			this.parse()
+				.done( this.onParseDone.bind( this) )
+				.fail( this.onParseFail.bind( this) );
+		}, this );
+	}
+	return ve.ui.WikiaSourceModeDialog.super.prototype.getActionProcess.call( this, action );
+};
+
+ve.ui.WikiaSourceModeDialog.prototype.parse = function () {
+	return $.ajax( {
+		url: mw.util.wikiScript( 'api' ),
+		data: {
+			action: 'visualeditor',
+			paction: 'parsefragment',
+			page: mw.config.get( 'wgRelevantPageName' ),
+			wikitext: this.sourceModeTextarea.getValue(),
+			token: mw.user.tokens.get( 'editToken' ),
+			format: 'json'
+		},
+		dataType: 'json',
+		type: 'POST',
+		// Wait up to 100 seconds before giving up
+		timeout: 100000,
+		cache: 'false'
+	} );
+};
+
+ve.ui.WikiaSourceModeDialog.prototype.onParseDone = function ( response ) {
+	if ( !response ||
+		response.error ||
+		!response.visualeditor ||
+		response.visualeditor.result !== 'success' ||
+		response.visualeditor.content === false ) {
+		return this.onParseFail.call( this );
+	}
+	this.close().done( function () {
+		var target = this.surface.getTarget();
+		target.deactivating = true;
+		target.toolbarSaveButton.disconnect( target );
+		target.toolbarSaveButton.$element.detach();
+		target.getToolbar().$actions.empty();
+		target.teardownSurface( true ).done( function () {
```

Comment:
This was changed so that true is needed again. I added in the noAnimate flag so that we weren't relying on the lack of CSS transitions to prevent an animation